feg/gateway/diameter/connection_manager.go

```diff
@@ -98,12 +112,45 @@ func (cm *ConnectionManager) DisableFor(period time.Duration) {
 
 // Enable - enables new connection creations
 func (cm *ConnectionManager) Enable() {
+	glog.V(2).Info("ConnectionManager: enabling connections")
 	cm.rwl.Lock()
 	defer cm.rwl.Unlock()
 	cm.disabled = false
 }
 
+// Find returns connection object matching given diameter connection or nil if not found
+func (cm *ConnectionManager) Find(dc diam.Conn) *Connection {
+	if cm == nil || dc == nil {
+		return nil
+	}
+	cm.rwl.RLock()
+	defer cm.rwl.RUnlock()
+	for _, conn := range cm.connMap {
+		if conn != nil {
+			if conn.conn == dc {
```

Comment:
not from this PR, but maybe it would be good to change the field `conn` on the Connection strunct and use  `diamCon`

```
type Connection struct {
	diamConn     diam.Conn         <- this
	metadata *smpeer.Metadata
	server   *DiameterServerConfig
	client   *sm.Client
	mutex    sync.Mutex
}
```

That confused me when I was reading the code